plugins/woocommerce/templates/checkout/order-received.php

```diff
@@ -32,10 +32,10 @@
 	 */
 	$message = apply_filters(
 		'woocommerce_thankyou_order_received_text',
-		__( 'Thank you. Your order has been received.', 'woocommerce' ),
+		esc_html( __( 'Thank you. Your order has been received.', 'woocommerce' ) ),
 		$order
 	);
 
-	echo esc_html( $message );
+	echo $message;
```

Comment:
We'll need a linting directive before this line: 
```suggestion
	// phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
	echo $message;
```